syntax = "proto3";

message modelRequest {
    int32 user = 1;
    string modelName = 2;
    string prompt = 3;
}

message modelResponse {
    string prompt = 1;
    string response = 2;
    float executionTime = 3;
    float totalTime = 4;
}

message modelName {
    string modelName = 1;
}

message modelConfig {
    string modelName = 1;
    int32 maxToken = 2;
    bool loadUserRecord = 3;
}

message modelInfo {
    string modelInfo = 1;
}

message userRecord {
    string userRecord = 1;
}

message queueSize {
    string queueSize = 1;
}

message currentUser {
    string currentUser = 1;
}

message empty {
}




service LLMService {
    rpc getModelResponse (modelRequest) returns (modelResponse);
    rpc deleteModelProc (modelName) returns (modelInfo);
    rpc createModelProc (modelConfig) returns (modelInfo);
    rpc checkModelState (empty) returns (modelInfo);
    rpc checkUSerRecord (modelName) returns (userRecord);
    rpc showQueurSize (modelName) returns (queueSize);
    rpc showCurrentUser (modelName) returns (currentUser);
}

